zerver/views/streams.py

```diff
@@ -885,6 +885,32 @@ def send_messages_for_new_subscribers(
                     history_public_to_subscribers=stream.history_public_to_subscribers,
                     is_web_public=stream.is_web_public,
                 )
+                new_channel_message = None
+
+                # Policy `public_protected_history` is missing here as those channels don't get
+                # channel creation notification.
+                if policy_key == "web_public":
+                    new_channel_message = _(
+                        "**Web-public** channel created by {user_name}. **Description:**\n"
+                        "{channel_description_in_quotes}"
+                    )
+                elif policy_key == "public":
+                    new_channel_message = _(
+                        "**Public** channel created by {user_name}. **Description:**\n"
+                        "{channel_description_in_quotes}"
+                    )
+                elif policy_key == "private_shared_history":
+                    new_channel_message = _(
+                        "**Private, shared history** channel created by {user_name}. **Description:**\n"
+                        "{channel_description_in_quotes}"
+                    )
+                elif policy_key == "private_protected_history":
+                    new_channel_message = _(
+                        "**Private, protected history** channel created by {user_name}. **Description:**\n"
+                        "{channel_description_in_quotes}"
+                    )
+
+                assert new_channel_message is not None
```

Comment:
I don't think the code you have written is getting used at all.

`new_channel_message` is never getting used & `channel_description_in_quotes` is never set, right ?